components/logins/src/encryption.rs

```diff
@@ -36,6 +36,19 @@ pub struct EncryptorDecryptor {
     jwk: jwcrypto::Jwk,
 }
 
+trait CryptoResultExt<T> {
+    fn map_crypto_err(self, action: &str) -> Result<T>;
+}
+
+impl<T> CryptoResultExt<T> for std::result::Result<T, jwcrypto::JwCryptoError> {
+    fn map_crypto_err(self, action: &str) -> Result<T> {
+        self.map_err(|e| Error::CryptoError {
+            action: action.into(),
+            error: e,
+        })
+    }
+}
+
```

Comment:
I like the [extend](https://crates.io/crates/extend) crate for things like this.  We use it in UniFFI, so I think it's already moz-central approved.  Maybe we should add it to app-services, what do you think?